src/models/services/rule-service.ts

```diff
@@ -0,0 +1,72 @@
+import { RuleInterface } from "../interfaces/configuration/rule-interface";
+import { ConfigurationInterface } from "../interfaces/configuration/configuration-interface";
+
+export class RuleService {
+    private rule?: RuleInterface;
+
+    constructor(config: ConfigurationInterface) {
+        this.rule = this.findRuleForCurrentDomain(config.rules ? config.rules : []);
+    }
+
+    /**
+     * Yields to true, if the chat may be displayed on the current domain.
+     * If false, this means that the chat may not be displayed on the current domain.
+     */
+    public mayDisplayForCurrentDomain(): boolean {
+        return this.rule == null || this.rule.disable !== true;
+    }
+
+    /**
+     * Returns the rule for the current domain.
+     */
+    private findRuleForCurrentDomain(rules: RuleInterface[]): RuleInterface | undefined {
+        const location = window.location;
+        const currentPathName = location.pathname.toLowerCase();
+        const currentHost = location.hostname.toLowerCase();
+        const currentPort = location.port;
+        const currentProtocol = location.protocol;
```

Comment:
```location.origin``` contains protocol, host and port in one string: "http://localhost:4444" "https://github.com", etc...